### PR TITLE
Add type definitions to activities

### PR DESCRIPTION
**Pull request by sfentress**
Mostly just a refactoring. There are very few functional changes, just a couple places where components may not have been handling the possibility of a prop being undefined.

### CODE REVIEW

File: src/components/activity-page/managed-interactive/managed-interactive.tsx [hunk 1]
```diff
@@ -1,12 +1,21 @@
...
 }
 
+interface x {
+  foo: string;
+}
+
+interface y {
+  bar: string;
+}
+
```
Comment (mklewandowski) [inline src/components/activity-page/managed-interactive/managed-interactive.tsx:18]: Probably meant to come out?

Comment (sfentress) [reply src/components/activity-page/managed-interactive/managed-interactive.tsx:18]: Ha, good catch.

That was just me confirming to my own satisfaction that we can't do

```
interface x {
  foo: string;
}

interface y {
  bar: string;
}

const getFooBar = (xOrY: x | y) {
  if (xOrY.foo) {
    ....
  }
}
```

which is really annoying...

Comment (sfentress) [reply src/components/activity-page/managed-interactive/managed-interactive.tsx]: I mean... that was a test to find out if anyone would really read through the whole commit.

**Comment by mklewandowski** [comment]
@sfentress are we also missing a definition for EmbeddableXhtmlInteractive?

**Comment by sfentress** [comment]
> Are we also missing a definition for EmbeddableXhtmlInteractive?

Is that something beyond

```
export interface Embeddable {
  type: "ManagedInteractive" | "MwInteractive" | "Embeddable::Xhtml";
  content?: string;
}
```

That's already a part of `Embeddable`. I was considering splitting it into

```
interface BaseEmbeddable { ... }

interface ManagedEmbeddable extends BaseEmbeddable { ...}
interface MwEmbeddable extends BaseEmbeddable { ...}
interface XhtmlEmbeddable extends BaseEmbeddable { ...}

type Embeddable = ManagedEmbeddable | MwEmbeddable | XhtmlEmbeddable
```
but it made the code in `embeddable.tsx` much more complicated. I could go back to it, though, if it's clearer.

If you mean something else, then I may just be missing it.

**Comment by mklewandowski** [comment]
I was looking specifically at text-box.tsx which is expecting EmbeddableXhtmlInteractive to be exported from the new types file.

**Comment by mklewandowski** [comment]
Ah, I'm assuming then that you meant for text-box.tsx to use `Embeddable` as the type on the prop.

**Comment by sfentress** [comment]
D'oh, mistaken commit from when I was testing that Embeddable type split.

File: src/components/activity-page/text-box/text-box.tsx [hunk 2]
```diff
@@ -2,9 +2,10 @@ import React from "react";
 import { renderHTML } from "../../../utilities/render-html";
 
 import "./text-box.scss";
+import { EmbeddableXhtmlInteractive } from "../../../types";
 
 interface IProps {
-  embeddable: any;
+  embeddable: EmbeddableXhtmlInteractive;
```
Comment (mklewandowski) [inline src/components/activity-page/text-box/text-box.tsx:8]: Meant to just be `Embeddable` as the type?

**Pull request merged**